atomic_reactor/plugins/post_export_operator_manifests.py

```diff
@@ -78,79 +53,53 @@ def run(self):
 
         :return: str, path to operator manifests zip file
         """
-        if not self.should_run():
+        if not (
+                has_operator_bundle_manifest(self.workflow) or
+                has_operator_appregistry_manifest(self.workflow)
+        ):
+            self.log.info("Operator manifests label not set in Dockerfile. Skipping")
             return
 
-        # OSBS2 TBD this should go in the context dir
-        manifests_archive_dir = tempfile.mkdtemp()
-#        image = self.workflow.image
-        # OSBS2 TBD
-        # just use oc image extract to get image content
-#        container_dict = self.tasker.create_container(image, command=['/bin/bash'])
-#        container_id = container_dict['Id']
-#        try:
-#            bits, _ = self.tasker.get_archive(container_id,
-#                                              IMG_MANIFESTS_PATH)
-#        except APIError as ex:
-#            msg = ('Could not extract operator manifest files. '
-#                   'Is there a %s path in the image?' % (IMG_MANIFESTS_PATH))
-#            self.log.debug('Error while trying to extract %s from image: %s',
-#                           IMG_MANIFESTS_PATH, ex)
-#            self.log.error(msg)
-#            raise RuntimeError('%s %s' % (msg, ex)) from ex
-#
-#        except Exception as ex:
-#            raise RuntimeError('%s' % ex) from ex
-#
-#        finally:
-#            try:
-#                self.tasker.remove_container(container_id)
-#            except Exception as ex:
-#                self.log.warning('Failed to remove container %s: %s', container_id, ex)
-        bits = b''
-
-        with tempfile.NamedTemporaryFile() as extracted_file:
-            for chunk in bits:
-                extracted_file.write(chunk)
-            extracted_file.flush()
-            tar_archive = tarfile.TarFile(extracted_file.name)
-
-        tar_archive.extractall(manifests_archive_dir)
-        manifests_path = os.path.join(manifests_archive_dir, MANIFESTS_DIR_NAME)
+        platforms = get_platforms(self.workflow.data)
+        image: ImageName = self.workflow.data.tag_conf.get_unique_images_with_platform(
+            platforms[0])[0]
+        tmp_dir = tempfile.mkdtemp(dir=self.workflow.build_dir.any_platform.path)
+        manifests_dir = os.path.join(tmp_dir, MANIFESTS_DIR_NAME)
+
+        self.workflow.imageutil.extract_file_from_image(image, IMG_MANIFESTS_PATH, tmp_dir)
 
         if has_operator_bundle_manifest(self.workflow):
-            self._verify_csv(manifests_path)
+            self._verify_csv(manifests_dir)
 
-        manifests_zipfile_path = os.path.join(manifests_archive_dir, OPERATOR_MANIFESTS_ARCHIVE)
+        manifests_zipfile_path = (self.workflow.build_dir.any_platform.path /
+                                  OPERATOR_MANIFESTS_ARCHIVE)
```

Comment:
Previously, we zipped the manifests that we extracted from the image. Now, we're taking the manifests from the build dir instead. Those should usually be identical, but we can't be sure (the only file we verify is the ClusterServiceVersion one).

Could we keep it as it was before and take the extracted ones?

Reply:
There seems to be some confusion here. We're taking it from extracted dir only(`tmp_dir`). It might be because I have named the extracted manifests directory as `manifest_dir`

Reply:
Oh, yeah, don't mind me, I mixed up which path is being `os.walk`-ed and which is being zipped :smile: